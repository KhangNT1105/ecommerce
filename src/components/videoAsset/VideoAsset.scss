@import '../../styles/variables.scss';
@import '../../styles/zIndex.scss';

.video-asset-block {
  .header {
    padding: 0.5rem 0;
    color: $footerColor;
    font-weight: 700;
  }

  .content-body {
    position: relative;
    width: 100%;
    height: 100%;
    padding-top: 56.125%;
    border: 1px solid $grey-border;
    border-top: 0;
    background-color: $grey-1;

    .content-wrapper {
      position: absolute;
      top: 0;
      left: 0;
      bottom: 0;
      right: 0;
      display: flex;
      justify-content: center;
      align-items: center;

      span {
        font-size: 1.75em;
        font-weight: bold;
        color: $grey;
      }
    }

    video {
      &:focus {
        outline: none;
      }
    }

    iframe {
      margin: 0;
      padding: 0;
      border: none;
    }
  }

  .input-group {
    input.form-control {
      padding: 1.25rem 8.065rem 1.25rem 1.25rem;
      border: 1px solid $grey-border;
      border-radius: 0;
      border-top-left-radius: 8px;
      border-top-right-radius: 8px;
    }

    .input-group-prepend,
    .input-group-append {
      position: absolute;
      top: 50%;
      right: 1.5rem;
      transform: translateY(-50%);
      z-index: $zIndexInputGroupAppend;

      button.btn {
        min-height: unset;
        padding: 0.1rem 0.5rem;
        border-radius: 4px;
        font-size: 0.8rem;
      }
    }
  }
}
