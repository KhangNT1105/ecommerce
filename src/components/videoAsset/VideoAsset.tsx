import React, { useState, useRef } from 'react';
import { Input, InputGroup, InputGroupAddon } from 'reactstrap';
import { useTranslation } from 'react-i18next';
import './VideoAsset.scss';
import Button from 'components/button/Button';

export type InputType = 'text' | undefined;
export type BtnType = 'button' | 'submit' | undefined;
export type VideoTag = 'video' | 'iframe';

interface inputProps {
  name: string;
  placeholder?: string;
  inputType?: InputType;
}

interface buttonProps {
  children: React.ReactNode;
  color?: string;
  btnType?: BtnType;
  onClick: (url: string) => void;
}

interface videoProps {
  src?: string;
  width?: string;
  height?: string;
  videoTag: VideoTag;
  titleVideo: string;
}

interface VideoAssetProps {
  inputProps: inputProps;
  buttonProps: buttonProps;
  titleChildren: React.ReactNode;
  videoProps: videoProps;
}

const VideoAsset: React.FC<VideoAssetProps> = ({ inputProps, buttonProps, titleChildren, videoProps }) => {
  const { t } = useTranslation();
  const [url, setUrl] = useState('');
  const { name, placeholder, inputType } = inputProps;
  const { children, color, btnType, onClick } = buttonProps;
  const { src, width, height, videoTag, titleVideo } = videoProps;
  const hasVideo = src ? true : false;
  const videoRef = useRef(null);

  const handleChange = (e: React.FormEvent<HTMLInputElement>) => {
    const value = e.currentTarget.value;
    setUrl(value);
  };

  const handleOnClick = async (e: React.FormEvent<HTMLButtonElement>) => {
    e.preventDefault();
    // Need to check url
    onClick(url);
  };

  return (
    <>
      <div className="video-asset-block">
        <div className="header" children={titleChildren} />
        <div className="content">
          <div className="content-header">
            <InputGroup>
              <Input name={name} placeholder={placeholder} type={inputType} onChange={handleChange} value={url} />
              <InputGroupAddon addonType="prepend">
                <Button children={children} color={color} type={btnType} onClick={handleOnClick} />
              </InputGroupAddon>
            </InputGroup>
          </div>

          <div className="content-body">
            <div className="content-wrapper">
              {hasVideo ? (
                videoTag === 'iframe' ? (
                  <iframe src={src} width={width} height={height} title={titleVideo} ref={videoRef} />
                ) : (
                  <video
                    title={titleVideo}
                    width={width}
                    height={height}
                    controls={true}
                    autoPlay={true}
                    src={src}
                    ref={videoRef}
                  />
                )
              ) : (
                <span>{t('VIDEO_PREVIEW')}</span>
              )}
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

export default VideoAsset;
