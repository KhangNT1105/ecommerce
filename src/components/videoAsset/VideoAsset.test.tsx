import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import VideoAsset from './VideoAsset';

describe('Video Asset Block', () => {
  test('renders correctly', () => {
    const handleOnClick = (url: string) => {};

    const inputProps = {
      name: 'video-url',
      placeholder: 'Enter URL here',
      inputType: 'text' as 'text'
    };

    const buttonProps = {
      children: 'Regenerate',
      color: 'success',
      btnType: 'button' as 'button',
      onClick: handleOnClick
    };

    const titleChildren = <span className="title">Asset 1 URL*:</span>;

    const videoProps = {
      src: 'http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4',
      width: '100%',
      height: '100%',
      videoTag: 'video' as 'video',
      titleVideo: 'Big Buck Bunny'
    };

    const wrapper = shallow(
      <VideoAsset
        inputProps={inputProps}
        buttonProps={buttonProps}
        titleChildren={titleChildren}
        videoProps={videoProps}
      />
    );

    expect(wrapper.find('.video-asset-block').exists()).toEqual(true);
  });

  test('renders snapshot correctly', () => {
    const handleOnClick = (url: string) => {};

    const inputProps = {
      name: 'video-url',
      placeholder: 'Enter URL here',
      inputType: 'text' as 'text'
    };

    const buttonProps = {
      children: 'Regenerate',
      color: 'success',
      btnType: 'button' as 'button',
      onClick: handleOnClick
    };

    const titleChildren = <span className="title">Asset 1 URL*:</span>;

    const videoProps = {
      src: 'http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4',
      width: '100%',
      height: '100%',
      videoTag: 'video' as 'video',
      titleVideo: 'Big Buck Bunny'
    };

    const tree = renderer
      .create(
        <VideoAsset
          inputProps={inputProps}
          buttonProps={buttonProps}
          titleChildren={titleChildren}
          videoProps={videoProps}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
