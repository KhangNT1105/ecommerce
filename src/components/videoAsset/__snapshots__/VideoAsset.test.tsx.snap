// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Asset Block renders snapshot correctly 1`] = `
<div
  className="video-asset-block"
>
  <div
    className="header"
  >
    <span
      className="title"
    >
      Asset 1 URL*:
    </span>
  </div>
  <div
    className="content"
  >
    <div
      className="content-header"
    >
      <div
        className="input-group"
      >
        <input
          className="form-control"
          name="video-url"
          onChange={[Function]}
          placeholder="Enter URL here"
          type="text"
          value=""
        />
        <div
          className="input-group-prepend"
        >
          <button
            aria-label={null}
            className="  btn btn-success btn-sm"
            onClick={[Function]}
            type="button"
          >
            Regenerate
          </button>
        </div>
      </div>
    </div>
    <div
      className="content-body"
    >
      <div
        className="content-wrapper"
      >
        <video
          autoPlay={true}
          controls={true}
          height="100%"
          src="http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4"
          title="Big Buck Bunny"
          width="100%"
        />
      </div>
    </div>
  </div>
</div>
`;
